c_tests/tests/inline_asm.c

```diff
@@ -0,0 +1,38 @@
+// Compiler:
+// Run-time:
+//   env-var: YK_PRINT_IR=1
+//   stderr:
+//     ...
+//     ...call i32 asm "mov $$5, $0"...
+//     ...
+
+// Check that we can handle inline asm properly.
+//
+// FIXME An optimising compiler can remove all of the code between start/stop
+// tracing.
+
+#include <assert.h>
+#include <stdlib.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  int res = 0;
+  void *tt = __yktrace_start_tracing(HW_TRACING, &res);
+  // Stores the constant 5 into `res`.
+  asm("mov $5, %0"
```

Comment:
I think we should guard this with something like:

```c
#if <x86>
asm(...)
#else
#error asm statement not tested on this platform
```

just so that if/when we port to another platform we still test this properly.

Reply:
Fixed in a0738bf